### ISSUE TITLE
illegal storage access when awaiting future inside call

### ISSUE THREAD

**Issue by blackmius** [yglukhov/yasync#2]
```nim
import yasync

proc a(): int {.async.} =
  return 1

proc b() {.async.} =
  echo(await a())

discard b()
```

```
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

**Comment by yglukhov** [comment]
That's nim crashing, not the program. It is a nim bug, which unfortunately reproduces only with yasync, when it's using closure-get-env-type hack under the hood, so i don't feel entitled to report it yet. It's easier to just use a temp variable as a workaround.

**Comment by yglukhov** [comment]
This is now fixed in yasync in 0d8adbf0971d6cad07b43a6da8e9397f3bdd0e61
